### PR TITLE
docs: refactor observables overview

### PR DESCRIPTION
**Pull request by jbogarthyde**
Refactor existing documentation to follow current style standards and use task-based organization

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Style: uses "we" construction
Language is sometimes awkward
Refers to things "discussed later" without explicit reference

Issue Number: N/A

## What is the new behavior?

Correct construction and language for style and clarity
Add ref to RxJS Subject doc for multicasting (from contributor suggestion).
Move error handling from RxJS page to this page (replacing forward reference)

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: aio/content/guide/observables.md [hunk 1]
```diff
@@ -1,18 +1,23 @@
...
 Observables provide support for passing messages between parts of your application.
-They are used frequently in Angular and are the recommended technique for event handling, asynchronous programming, and handling multiple values.
+They are used frequently in the Angular API, and are the recommended technique for event handling, asynchronous programming, and handling multiple values.
 
-The observer pattern is a software design pattern in which an object, called the *subject*, maintains a list of its dependents, called *observers*, and notifies them automatically of state changes.
-This pattern is similar (but not identical) to the [publish/subscribe](https://en.wikipedia.org/wiki/Publish%E2%80%93subscribe_pattern) design pattern.
+## How observables work
+
+The observer pattern is a software design pattern in which an object (sometimes called the *subject*) maintains a list of its dependents, called *observers*, and notifies them automatically of state changes.
+This pattern is similar (but not identical) to the [publish/subscribe](https://en.wikipedia.org/wiki/Publish%E2%80%93subscribe_pattern "Wikipedia") design pattern.
```
Comment (rockument69) [inline aio/content/guide/observables.md:9]: Remove this parenthetical unless we explain the diffs, which is probably not necessary. Use of "similar" is good enough IMHO.

File: aio/content/guide/observables.md [hunk 2]
```diff
@@ -1,18 +1,23 @@
 # Using observables to pass values
 
 Observables provide support for passing messages between parts of your application.
```
[the remark this answers is not in the document]
Comment (rockument69) [reply aio/content/guide/observables.md:3]: The "handling multiple values" part comes later. Probably don't need it here anyway.

File: aio/content/guide/observables.md [hunk 3]
```diff
@@ -1,18 +1,23 @@
...
+This pattern is similar (but not identical) to the [publish/subscribe](https://en.wikipedia.org/wiki/Publish%E2%80%93subscribe_pattern "Wikipedia") design pattern.
 
 Observables are declarative&mdash;that is, you define a function for publishing values, but it is not executed until a consumer subscribes to it.
-The subscribed consumer then receives notifications until the function completes, or until they unsubscribe.
+The subscribed consumer then receives notifications until the function completes or errors out, or until they unsubscribe.
 
 An observable can deliver multiple values of any type&mdash;literals, messages, or events, depending on the context. The API for receiving values is the same whether the values are delivered synchronously or asynchronously. Because setup and teardown logic are both handled by the observable, your application code only needs to worry about subscribing to consume values, and when done, unsubscribing. Whether the stream was keystrokes, an HTTP response, or an interval timer, the interface for listening to values and stopping listening is the same.
```
Comment (rockument69) [inline aio/content/guide/observables.md:14]: This section could use some editing.

File: aio/content/guide/observables.md [hunk 4]
```diff
@@ -1,18 +1,23 @@
...
 
 Because of these advantages, observables are used extensively within Angular, and are recommended for app development as well.
 
+Angular makes use of and extends [RxJS](https://rxjs.dev/guide/overview "RxJS documentation") (Reactive Extensions for JavaScript), a library for [reactive programming](https://en.wikipedia.org/wiki/Reactive_programming "Wikipedia").
```
Comment (rockument69) [inline aio/content/guide/observables.md:18]: I get lost here. Do "I" use RxJS, or does Angular? If only the latter, why would "I" need to know?

File: aio/content/guide/observables.md [hunk 5]
```diff
@@ -1,18 +1,23 @@
...
+RxJS provides an implementation of the `Observable` type, which makes it easier to compose asynchronous or callback-based code in TypeScript applications, as well as many useful [operators and utility functions](guide/rx-library "Using RxJS functions and operators").
```
Comment (rockument69) [inline aio/content/guide/observables.md:19]: This is really what RxJS is about. It makes it easier for "you to compose asynchronous or callback-based code in your app."

File: aio/content/guide/observables.md (hunk 4 above)
Comment (rockument69) [inline aio/content/guide/observables.md:18]: Before sending the reader to the RxJS site, we should send them to our page about RxJS. On _that_ page, this link makes sense.

File: aio/content/guide/rx-library.md [hunk 6]
```diff
@@ -1,14 +1,13 @@
 # The RxJS library
```
Comment (rockument69) [inline aio/content/guide/rx-library.md:1]: Need better title, one without the tongue-twisting and intimidating "RxJS" . How about "Creating and working with observables"? And "makes it easier to compose asynchronous or callback-based code" should probably be in the first sentence. I would use "Reactive Extensions for JavaScript" and put "RxJS" in parentheses.

File: aio/content/guide/observables.md (hunk 4 above)
Comment (rockument69) [reply aio/content/guide/observables.md:18]: After reading the RxJS page, my suggestion: Don't lead with the tongue-twisting and intimidating "RxJS" . I would use "Reactive Extensions for JavaScript" and put "RxJS" in parentheses.

Comment (rockument69) [reply aio/content/guide/observables.md:18]: You can then link to "Creating and working with observables" (the suggested title for the RxJS page).

**Review by rockument69 — COMMENTED** [review]
I have not yet had time to look at the other pages in the section. There may be ways to retitle some of them to make them clearer. I will add some comments to the [doc plan](https://docs.google.com/document/d/1l6Ov53GaFNW_6GqqJthTcBRv_JQMYmXk_jNFTiGr7PY/edit?usp=sharing).

File: aio/content/guide/observables.md (hunk 4 above)
Comment (jbogarthyde) [reply aio/content/guide/observables.md:18]: I'm taking your suggestion of combining that page into this one.

File: aio/content/guide/observables.md (hunk 5 above)
Comment (jbogarthyde) [reply aio/content/guide/observables.md:19]: Moving this up

**Comment by jbogarthyde** [comment]
This got munged, starting over with https://github.com/angular/angular/pull/38165

**Pull request closed without merging**